tools/rustfmt/srcs/main.rs

```diff
@@ -0,0 +1,207 @@
+use std::collections::VecDeque;
+use std::env;
+use std::fs;
+use std::path::{Path, PathBuf};
+use std::process::{Command, Stdio};
+use std::str;
+
+fn main() {
+    // Gather all command line and environment settings
+    let options = parse_args();
+
+    // Gather a list of all formattable targets
+    let targets = query_rustfmt_targets(&options);
+
+    // Run rustfmt on these targets
+    apply_rustfmt(&options, &targets);
+}
+
+/// A set of supported rules for use in a `bazel query` when querying for Rust targets.
+const SUPPORTED_RULES: &str =
+    "rust_library|rust_proc_macro|rust_shared_library|rust_static_library|rust_binary|rust_test";
+
+/// Perform a `bazel` query to determine a list of Bazel targets which are to be formatted.
+fn query_rustfmt_targets(options: &Config) -> Vec<String> {
+    let scope = match &options.package {
+        Some(target) => {
+            if !target.ends_with(":all") && !target.ends_with("...") {
+                return vec![target.to_string()];
+            }
+            target
+        }
+        None => "//...:all",
+    };
+
+    let query_args = vec![
+        "query".to_owned(),
+        format!(
+            r#"kind('{types}', {scope}) except attr(tags, 'norustfmt|manual', kind('{types}', {scope}))"#,
```

Comment:
I'm not sure about removing `manual` targets? Wouldn't we still want to `rustfmt` e.g. a test marked manual?

Reply:
In order for the `rustfmt_aspect` to work it would need a buildable target. If a user wants to format that target, then to respect the `manual` tag, they should explicitly format that target.

Reply:
the `manual` tag has been removed.